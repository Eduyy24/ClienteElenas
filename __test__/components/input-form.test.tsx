import React from 'react';
import { fireEvent, render } from '@testing-library/react-native'
import InputForm from '../../src/components/input-form';


describe('<InputForm />', () => {

  test('Comprobar la UI del componente ButtonFlex', () => {
    const input = <InputForm onChangeText={() => {}} label="Nombre" />
    const {toJSON} = render(input);
    expect(toJSON()).toMatchSnapshot();
  });

  test('se valida el llamado al callback, cuando se escribe en el input, se genera snapshot del comportamiento', async () => {
    const onPress = jest.fn()
    const inputForm = <InputForm onChangeText={onPress} label="Nombre" />
    const wraper = render(inputForm)

    const input = wraper.getByTestId('input')
    fireEvent.changeText(input, 'Eduardo')

    expect(onPress).toHaveBeenCalledTimes(1);
    expect(wraper.toJSON()).toMatchSnapshot();
  })
});