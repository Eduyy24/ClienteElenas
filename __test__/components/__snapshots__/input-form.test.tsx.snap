// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputForm /> Comprobar la UI del componente ButtonFlex 1`] = `
<View
  style={
    Object {
      "height": 90,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "marginVertical": 10,
      }
    }
  >
    Nombre
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#A2A7A8",
        "borderRadius": 15,
        "height": "40%",
        "maxHeight": 50,
        "minHeight": 30,
        "paddingHorizontal": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#fff",
          "flex": 1,
        }
      }
      testID="input"
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`<InputForm /> se valida el llamado al callback, cuando se escribe en el input, se genera snapshot del comportamiento 1`] = `
<View
  style={
    Object {
      "height": 90,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "marginVertical": 10,
      }
    }
  >
    Nombre
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#A2A7A8",
        "borderRadius": 15,
        "height": "40%",
        "maxHeight": 50,
        "minHeight": 30,
        "paddingHorizontal": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={
        [MockFunction] {
          "calls": Array [
            Array [
              "Eduardo",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#fff",
          "flex": 1,
        }
      }
      testID="input"
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
